src/API/Reports/Products/DataStore.php

```diff
@@ -526,4 +527,90 @@ protected function initialize_queries() {
 		$this->subquery->add_sql_clause( 'from', self::get_db_table_name() );
 		$this->subquery->add_sql_clause( 'group_by', 'product_id' );
 	}
+
+	/**
+	 * Possible add filter to insert product template.
+	 */
+	public static function possibly_add_product_template() {
+		if ( isset( $_GET['post_type'] ) && 'product' === wc_clean( wp_unslash( $_GET['post_type'] ) ) && isset( $_GET['product_template'] ) ) { // phpcs:ignore CSRF ok, sanitation ok.
+			// default_excerpt is last filter called in get_default_post_to_edit.
+			add_filter( 'default_excerpt', array( __CLASS__, 'insert_product_template' ), 10, 2 );
+			// add_action('wp_insert_post', array(__CLASS__, 'insert_product_template_action'), 10, 2);
+		}
+	}
+
+	/**
+	 * Loads the product template by the 'product_template' param, and updates the new post with the template data.
+	 *
+	 * @param string  $excerpt Default post excerpt.
+	 * @param WP_Post $post    Post object.
+	 * @return string $excerpt
+	 */
+	public static function insert_product_template_action( $excerpt, $post ) {
+		if ( 'product' === $post->post_type && 'auto-draft' === $post->post_status && ! empty( $_REQUEST['product_template'] ) ) {
+			remove_action('wp_insert_post', array(__CLASS__, 'insert_product_template_action'), 10, 2);
+			include_once WC_ADMIN_ABSPATH . 'src/ProductCSVTemplateImporter.php';
+			$template_name = wp_unslash( $_REQUEST['product_template'] ); // phpcs:ignore csrf ok, sanitization ok.
+			$template_path = __DIR__ . '/Templates/' . $template_name . '.csv';
+			$template_path = apply_filters( 'woocommerce_product_template_csv_file_path', $template_path, $template_name );
+
+			if ( file_exists( $template_path ) && class_exists( '\Automattic\WooCommerce\Admin\ProductCSVTemplateImporter' ) ) {
+				$importer_class = apply_filters( 'woocommerce_product_csv_importer_template_class', '\Automattic\WooCommerce\Admin\ProductCSVTemplateImporter' );
+				$args           = array(
+					'parse' => true,
+				);
+
+				$importer = new $importer_class( $template_path, $args );
+				$importer->update_product( $post->ID );
+
+				add_filter('default_content', function($content, $post) {
```

Comment:
I'd recommend sticking to non-anonymous functions for filter callbacks wherever possible. It's probably best to have these filters ensure that they only run once by removing themselves upon invocation (like the parent function here does). While it's unlikely that `get_default_post_to_edit()` will get called again before the end of this PHP request, it's certainly not impossible.

Reply:
Yeah I didn't like making use of these either, and it was kind of a back up if we wanted to go this route.